app/logic/validator_score_v1_logic.rb

```diff
@@ -142,6 +142,13 @@ def assign_block_and_vote_scores
       vote_distance_all_average = array_average(vote_distance_all)
       vote_distance_all_median = array_median(vote_distance_all)
 
+      p.payload[:this_batch].update(
```

Comment:
`update!` probably makes more sense here because you aren't specifically checking if the update succeeded or not. Otherwise we don't know if the update actually succeeded or not whereas `update!` will raise an exception and an error bubble up which I'll find out about right away.